Formula/f/flight_sql.rb

```diff
@@ -0,0 +1,28 @@
+class FlightSql < Formula
+  desc "Apache Arrow Flight SQL Server - with DuckDB and SQLite back-ends"
+  homepage "https://arrow.apache.org/docs/format/FlightSql.html"
+  url "https://github.com/voltrondata/flight-sql-server-example.git",
+      tag:      "v1.1.19",
+      revision: "96f9831a901c4660c9cfb6cf299a682ef9614bb2"
+  license "Apache-2.0"
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on "openssl@3" => :build
+  depends_on xcode: :build
+  depends_on :macos
+
+  def install
+    mkdir "build"
+    cd "build" do
+      system "cmake", "-S", "..", *std_cmake_args, "-G", "Ninja"
+      system "cmake", "--build", ".", "--target", "install"
+      bin.install "flight_sql"
+    end
+  end
+
+  test do
+    assert_equal "Flight SQL Server CLI: v1.1.19", shell_output("#{bin}/flight_sql --version").strip
+  end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?
Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
ok - thanks, @bevanjkay - I'll build a better test...

Reply:
This app runs a server until it is terminated manually by the user.  In the upstream Docker build CI (this [line](https://github.com/voltrondata/flight-sql-server-example/blob/96f9831a901c4660c9cfb6cf299a682ef9614bb2/Dockerfile.ci#L95) - I run this script: https://github.com/voltrondata/flight-sql-server-example/blob/main/scripts/test_flight_sql.sh - I can try to iterate on getting that to work in the brew formula, but I'm a bit over my head already :). I will try, however...

Reply:
I have updated the test to truly test the functionality of the server and client as requested.